src/svgdom/util/element_by_id_finder.cpp

```diff
@@ -0,0 +1,114 @@
+#include "element_by_id_finder.hpp"
```

Comment:
by the way, why did you choose that long name? Maybe we could make the name a bit shorter, because it is self evident that we are searching for `element`s, thus it is not needed to have the 'element' word in the name. Also, it makes the namings more inconsistent, as in case of class we need to indicate that more than one element can be found, which is also self evident and is also already indicated by the `find()` method's signature. This is why I proposed the `finder_by_id`, `find_by_class`, `find_by_tag` names. Do those names sound unnatural to native English speaker? Maybe it could be also `by_id_finder` but I think it is better to use Polish notation, when we first type the most general entity, i.e. `finder` and then qualify it more by adding `by_id`, but I'm not sure, at least to my ear it sounds more clear in this case.

Reply:
I used those name to be consistent with the Javascript methods, and denote that `id` returns a single element, and `class-name` and `tag-name` return one or more elements. I though about using `find_element()` and `find_elements()` as an alternative.

Reply:
Well, I don't think we need to  blindly follow Javascript namings here, C++ is a different language and has its specifics.
Javascript is dynamically typed, so there is no other way to indicate the return type of the method, other than saying that in method's name. In C++ we have strict types, so from the method signature we see what type it returns, if it is supposed to return multiple elements, then we see the return type as some container, e.g. `std::vector`, there is no chance to mix it with something else, as compiler will point you to the mistake in compile time (thanks to strict typing). So, adding notion to method's name is redundant information. Also, to avoid unnecessary typing we need to keep names as short as possible, while preserving evidence of what method/class does. So, as I mentioned, it is very unlikely that in context of `svgdom` we'd be searching for something else than `element`s, so I see no need to include the `element` word in the namings. Same about `tag`/`class` vs `tag_name`/`class_name`, the `name` part adds no value to the descriptiveness of the method, when one says just `tag` it is quite clear that we mean it's name, and not, for example length of the tag.
So, I still would like the names would be `finder_by_id`, `finder_by_class`, `finder_by_tag`. Could you do the renamings please?